packages/dev/gui/src/2D/controls/container.ts

```diff
@@ -619,6 +619,38 @@ export class Container extends Control {
         this._measureForChildren.copyFrom(this._currentMeasure);
     }
 
+    protected _cacheFullyDefinedDim: { width?: boolean; height?: boolean } = {
+        width: undefined,
+        height: undefined,
+    };
+
+    protected _getAdaptDimTo(dim: "width" | "height"): boolean {
+        if (dim === "width") {
+            return this.adaptWidthToChildren;
+        } else {
+            return this.adaptHeightToChildren;
+        }
+    }
+
+    public isDimensionFullyDefined(dim: "width" | "height"): boolean {
+        const hasChildDirty = this.children.some((c) => c.isDirty);

```

Comment:
I think it's wrong, it should check all children, not just direct descendants.

At this point, and now that we have a flag that prevents this code from being called if desired, I don't think we should maintain a cache, and just do the checks every time `isDimensionFullDefined` is called. Maintaining the cache isn't really worth it.